### PR TITLE
Refine store topo

### LINKED ISSUE #870: Store topology does not work well in K8s environment
## Bug Report

Please answer these questions before submitting your issue. Thanks!

**What did you do?**

Open store topology

**What did you expect to see?**

A nice topology view.

**What did you see instead?**

Labels are overlapped so that they are hard to see:

image

**What version of TiDB Dashboard are you using (`./tidb-dashboard --version`)?**

v4.0.9.

### PR DESCRIPTION
**Pull request by baurine**
close #870 

What did:
1. remove the duplicated prefix and suffix string for the label values that are too long
2. show tooltip for the truncated strings
3. show the label name and label value in 2 lines instead of the same line

Before:
image

After:
image

image

How to test:

```shell
$ cd ui
$ yarn storybook
```

### CODE REVIEW

**Review by breeswish — APPROVED** [review]
The effect looks good to me. I didn't review the code though. @shhdgit PTAL

**Comment by shhdgit** [comment]
OK. Let me familiarize myself with this.

**Comment by shhdgit** [comment]
By the way, github1s is too good to use when review.
image

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx [hunk 1]
```diff
@@ -93,6 +282,9 @@ export default function StoreLocationTree({
           `translate(${t.x + margin.left}, ${t.y + margin.top}) scale(${t.k})`
         )
 
+        // don't show tooltip if zoom or drag
+        enableTooltip.current = t.x === 0 && t.y === 0 && t.k === 1
```
Comment (shhdgit) [inline ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:286]: Can this be put in `zoom.on('end')` without expressions?

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx [hunk 2]
```diff
@@ -9,12 +9,197 @@ import { Space } from 'antd'
...
+// output:
+// {
+//   "aaa-111a.abc.123":"111a",
+//   "aaa-222a.abc.123":"222a",
+//   "aaa-333a.abc.123":"333a"
+// }
+export function getArrShortStrMap(strArr: string[]): ShortStrMap {
+  const shortStrMap: ShortStrMap = {}
+  const strSet = new Set(strArr)
+  if (strSet.size < 2) {
```
Comment (shhdgit) [inline ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:78]: Is this mean that there is no need to compress info when there is only one node.

Comment (baurine) [reply ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:78]: yep, you can't calculate the duplicated prefix or suffix for only one item.

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx (hunk 1 above)
Comment (baurine) [reply ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:286]: Good catch! let me try it.

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.module.less [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  text-align: center;
+  line-height: @line-height-base;
+  border-radius: @border-radius-base;
+  z-index: 10;
+  transition: all 0.1s ease-out;
+  pointer-events: none;
+  box-shadow: @box-shadow-base;
+}
+
+.tooltip:before {
```
[the remark this answers is not in the document]
Comment (baurine) [reply ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.module.less:19]: yep, will update it.

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx [hunk 4]
```diff
@@ -168,38 +394,69 @@ export default function StoreLocationTree({
...
           }
           return cyan[5]
         })
         .attr('stroke-width', 3)
 
-      // text for non-leaf nodes
+      // text for root node
       nodeEnter
-        .filter((d: any) => d._children !== undefined)
+        .filter(({ data: { name } }: any) => name === 'Stores')
```
Comment (shhdgit) [inline ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:406]: Maybe we need a variable for root node's name.

**Comment by shhdgit** [comment]
Sorry, I learned that I can use pending review. That can get better review experience.

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx [hunk 5]
```diff
@@ -168,38 +394,69 @@ export default function StoreLocationTree({
...
-        .attr('stroke-linejoin', 'round')
-        .attr('stroke-width', 3)
-        .attr('stroke', 'white')
+        .text(({ data: { name } }: any) => name)
+
+      // text for non-root and non-leaf nodes
+      const middleNodeText = nodeEnter
+        .filter(
+          ({ data: { name } }: any) =>
+            name !== 'Stores' && name !== 'TiFlash' && name !== 'TiKV'
```
Comment (shhdgit) [inline ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:417]: Same with TiKV, TiFlash node.

File: ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx (hunk 2 above)
Comment (shhdgit) [reply ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:78]: Got it when I read the code behind. Maybe the function name can simply describe the algorithm, or the algorithm can have its own function.

Comment (baurine) [reply ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:78]: how about `trimDuplicate`? do you have any better name?

Comment (shhdgit) [reply ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx:78]: I think `trimDuplicate` is good! My imagination is poor. :D

**Pull request merged**